lib/IRGen/IRGenSIL.cpp

```diff
@@ -1932,6 +1936,25 @@ static llvm::Value *createStringValAddr(IRGenModule &IGM, StringRef strVal) {
   return opNameValAddr;
 }
 
+template <typename SrcT>
+void IRGenSILFunction::createArrayAndSize(
+    ArrayRef<SrcT> elements, bool isFloat, const char *arrayName,
+    std::function<llvm::Value *(SrcT elt)> convertElt, Address *array,
+    llvm::Value **numElts) {
+  *numElts = llvm::ConstantInt::get(IGM.Int32Ty, elements.size());
+  *array = isFloat ? createAlloca(IGM.FloatTy, *numElts,
+                                  IGM.getPointerAlignment(), "tensorEltVals")
+                   : createAlloca(IGM.Int64Ty, *numElts,
+                                  IGM.getPointerAlignment(), "tensorEltVals");
+  for (unsigned idx = 0; idx < elements.size(); ++idx) {
```

Comment:
Use an intermediate variable to avoid recomputing `.size()`. 
Or simply `for (auto idx : range(elements.size()))`.

Reply:
done